test/framework/easyconfig.py

```diff
@@ -1204,6 +1203,49 @@ def test_dump_extra(self):
 
         dumped_ec = EasyConfig(testec)
 
+    def test_dump_template(self):
+        """ Test EasyConfig's dump() method for files containing templates"""
+        rawtxt = '\n'.join([
+            'easyblock = "EB_toy"',
+            '',
+            'name = "Foo"',
+            'version = "0.0.1"',
+            '',
+            'homepage = "http://foo.com/"',
+            'description = "foo description"',
+            '',
+            "toolchain = {'version': 'dummy', 'name': 'dummy'}",
+            '',
+            "sources = ['foo-0.0.1.tar.gz']",
+            '',
+            'preconfigopts = "--opt1=%s" % name',
+            'configopts = "--opt2=0.0.1"',
+            '',
+            "sanity_check_paths = {'files': ['files/foo/bar'], 'dirs':[] }",
+        ])
+
+        handle, testec = tempfile.mkstemp(prefix=self.test_prefix, suffix='.eb')
+        os.close(handle)
+
+        ec = EasyConfig(None, rawtxt=rawtxt)
+        ec.enable_templating = True
+        ec.dump(testec)
+        ectxt = read_file(testec)
+
+        self.assertTrue(ec.enable_templating)
+
+        patterns = [
+            r"sources = \['SOURCELOWER_TAR_GZ'\]",
+            r'description = "foo description"',  # no templating for description
```

Comment:
and check that `name = 'Foo'` isn't templated to `name = '%(name)s'`, since that wouldn't work :)